src/raftstore/store/worker/apply.rs

```diff
@@ -960,26 +958,31 @@ impl ApplyDelegate {
                     .with_label_values(&["add_peer", "all"])
                     .inc();
 
-                if exists {
-                    error!(
-                        "{} can't add duplicated peer {:?} to region {:?}",
-                        self.tag, peer, self.region
-                    );
-                    return Err(box_err!(
-                        "can't add duplicated peer {:?} to region {:?}",
-                        peer,
-                        self.region
-                    ));
+                let mut exists = false;
+                if let Some(p) = util::find_mut_peer(&mut region, store_id) {
+                    exists = true;
+                    if !p.get_is_learner() {
+                        error!(
+                            "{} can't add duplicated peer {:?} to region {:?}",
+                            self.tag, peer, self.region
+                        );
+                        return Err(box_err!(
+                            "can't add duplicated peer {:?} to region {:?}",
+                            peer,
+                            self.region
+                        ));
+                    } else {
+                        p.set_is_learner(false);
+                    }
+                }
+                if !exists {
+                    // TODO: Do we allow adding peer in same node?
```

Comment:
For now, we do not allow same peer in same node, can we remove it?

Reply:
I agree. @BusyJay please also check it.

Reply:
Not yet. It may in the future.